### ISSUE TITLE
loss is always 1

### ISSUE THREAD

**Issue by Terminal-K** [RazvanDu/DUCK-Net#6]
Hi, when I run the notebook for training, I find the loss is always 1, is this normal?
![Snipaste_2023-11-07_17-33-23](https://github.com/RazvanDu/DUCK-Net/assets/59523614/cf7e06d4-5ee2-43d5-9930-5e1c6df027cd)
By the way, I couldn't install tensorflow2.10.0 for some reason with the server, so I used tensorflow2.3.0 instead and also made the code running.

**Comment by maenstru56** [comment]
Hello,

This is not normal behaviour, the loss function should be decreasing during training. To fix the issue, please try the following;
- Try to install Tensorflow 2.10.0, even if the code runs with version 2.3.0 there might be some important differences in how the backend functions are implemented that are causing this issue
- Check the input and output images and their format, and make sure the network receives valid input 
- Check if the network actually learns during training. Maybe you can visualize some of the input and predicted images after every epoch to see how they compare with the ground truth.

I hope this helps!

**Comment by Terminal-K** [comment]
@maenstru56 
Hi, I think I have identified where the problem lies. When reading the mask image, I set 'as_grey' to True in the imread function as #4 , which causes the mask values to be in the range of (0,1), leading to an error (no raise) in the judgment of 'mask_'. I suggest modifying 
https://github.com/RazvanDu/DUCK-Net/blob/44eca89e5ce79aedac805b09d0b38e545ca696de/ImageLoader/ImageLoader2D.py#L54-L57
as follows:
`mask[mask_[:, :, 0] > 127] = 1`
and it can harness the features of NumPy for acceleration as well.

I successfully made the code work using this approach with the environment of tf2.3.0.
![image](https://github.com/RazvanDu/DUCK-Net/assets/59523614/9b0abf86-63bc-4893-b6cd-e7e62ed3d914)

**Comment by maenstru56** [comment]
Hello,

In our original data loader the "imread" function is called without the "as_grey=True" parameter, so the original code should work as intended.

Thanks for pointing out the issue and finding a fix when the "as_grey" parameter is set to "True"!
